packages/event-status-service/src/controllers/automation/create.ts

```diff
@@ -0,0 +1,53 @@
+import { Response } from 'express'
+import uuidv4 from 'uuid/v4'
+import { logger } from '../../core/logger'
+import { AppRequest, stringifyError } from '@reapit/node-utils'
+import { db } from '../../core/db'
+import { Automation, generateAutomationItem } from '../../schemas/automation.schema'
+
+type Payload = {
+  clientCode: Automation['clientCode']
+  messageChannel: Automation['messageChannel']
+  messageBody: Automation['messageBody']
+  triggerOnEventType: Automation['triggerOnEventType']
+}
+
+export default async (req: AppRequest, res: Response) => {
+  const payload = req.body as Payload
+  const { traceId } = req
+
+  try {
+    logger.info('Create new automation...', { traceId, payload })
+
+    if ((req as any).user.clientCode !== payload.clientCode) {
```

Comment:
Feels like `req.user.clientCode` would be better as `req.user?.clientCode` same with `payload?.clientCode` just to avoid potential run time errors